test/python/test_onedim.py

```diff
@@ -1133,6 +1132,51 @@ def test_equivalence_ratio(self):
         assert phi[0] == np.inf
         assert np.isclose(phi[-1], 0.0)
 
+    def test_two_point_control(self):
+        """
+        Computes a solution using a standard counterflow diffusion flame, and then
+        recomputes the solution with two-point control enabled. No temperature decrement is
+        used for the two-point control, so the solutions should be the same.
+        """
+        p = ct.one_atm
+        fuel='H2:1.0, AR:1.0'
+        T_fuel=300
+        mdot_fuel=0.05
+        oxidizer='O2:0.21, AR:0.78'
+        T_ox=300
+        mdot_ox=0.1
+        width=0.05
+
+        # Solution object used to compute mixture properties
+        gas = ct.Solution("h2o2.yaml")
+        gas.TP = T_fuel, p
+
+        # Flame object
+        sim = ct.CounterflowDiffusionFlame(gas, width=width)
+
+        # Set properties of the fuel and oxidizer mixtures
+        sim.fuel_inlet.mdot = mdot_fuel
+        sim.fuel_inlet.X = fuel
+        sim.fuel_inlet.T = T_fuel
+
+        sim.oxidizer_inlet.mdot = mdot_ox
+        sim.oxidizer_inlet.X = oxidizer
+        sim.oxidizer_inlet.T = T_ox
+
+        sim.solve(loglevel=2)
+        temperature_1 = sim.T
+
+        sim.two_point_control_enabled = True
+        spacing = 0.95
+        control_temperature = np.min(sim.T) + spacing * (np.max(sim.T) - np.min(sim.T))
+        sim.set_left_control_point(control_temperature)
+        sim.set_right_control_point(control_temperature)
+        sim.solve(loglevel=2)
+        temperature_2 = sim.T
+
+        # Check difference between the two solutions
+        self.assertArrayNear(temperature_1, temperature_2, rtol=1e-4, atol=1e-6)
+
```

Comment:
A couple of other tests I'd suggest (some of which can be done in conjunction with the above, some of which might make more sense as separate tests):
- Check that the parameters are retained correctly when using `Sim1D.save` and `Sim1D.restore`. For speed, this can be done on a flame where the profile is just the initial guess, without calling `solve`.
- Check some of the error conditions, like setting parameters without two-point control enabled, activating two-point control on flame types where it doesn't apply, or setting the control points to temperatures that aren't in the domain.
- Check all the property getter/setters

Reply:
Should I look to these reference file examples for how to output a solution during a test and read that solution in?

Reply:
I figured it out.